journal/src/main/java/io/camunda/zeebe/journal/file/SegmentedJournalWriter.java

```diff
@@ -83,10 +83,10 @@ void append(final JournalRecord record) {
     }
   }
 
-  void append(final long checksum, final byte[] serializedRecord) {
+  JournalRecord append(final long checksum, final byte[] serializedRecord) {
```

Comment:
:upside_down_face: IntelliJ flags this has a 15 line of duplicated code (mainly the logic of creating a new segment and appending there on error). Would be a good opportunity to simplify here. I kind of wish we could also just use a functional approach with the `Either` type, but this would require us extending it with `flatMapLeft`, so I'd postpone this for now and just do some deduplication :)

Reply:
:upside_down_face: I was too lazy to re-write it to remove the duplication. Now that you called out, I have no other choice :laughing:

Reply:
Finally, I'm not the only lazy one :smile: